### PR TITLE
feat(PPDSC-2107): add tracking support to structured list and text field

### PR DESCRIPTION
**Pull request by evgenitsn**
PPDSC-2107

**What**

Add tracking support for:
- `onClick` on `<StructuredList/>` 
- `onFocus` on `<TextField/>` 

**I have done:**
- [x] Written unit tests against changes
- [ ] Written functional tests against the component and/or NewsKit site
- [x] Updated relevant documentation

**I have tested manually:**
- [ ] The feature's functionality is working as expected on Chrome, Firefox, Safari and Edge
- [ ] The screen reader reads and flows through the elements as expected.
- [x] There are no new errors in the browser console coming from this PR.
- [ ] When visual test is not added, it renders correctly on different browsers and mobile viewports (Safari, Firefox, small mobile viewport, tablet)
- [ ] The Playground feature is working as expected

**Who should review this PR:**

**How to test:**

### CODE REVIEW

File: src/structured-list/structured-list-item.tsx [hunk 1]
```diff
@@ -143,6 +155,16 @@ const ThemelessStructuredListItem = React.forwardRef<
             href={href}
             data-testid="list-item-link"
             aria-disabled={disabled && 'true'}
+            onClick={() => {
+              fireEvent({
+                originator: eventOriginator,
+                trigger: EventTrigger.Click,
+                context: {
+                  href,
+                  ...eventContext,
```
Comment (Vanals) [inline src/structured-list/structured-list-item.tsx:164]: is the href needed? Just wondering what the evntContext already containg

**Comment by Vanals** [comment]
Just left a little comment bu LGTM, 👍

File: src/structured-list/structured-list-item.tsx (hunk 1 above)
Comment (evgenitsn) [reply src/structured-list/structured-list-item.tsx:164]: Not really if it's needed but I followed [this example](https://github.com/newscorp-ghfb/newskit/blob/634832f55c88d33f63f1220d62ec3ebe4ed0dfbb/src/link/internal-link.tsx#L76-L77) which is again _**a link**_ and I decided to add it here as well.

The `eventContext` is `undefined` by default and can accept an object with user defined data.

File: src/text-field/text-field.tsx [hunk 2]
```diff
@@ -19,21 +20,32 @@ const ThemelessTextField = React.forwardRef<HTMLInputElement, TextFieldProps>(
       onBlur,
       onFocus,
       onChange,
+      eventContext,
```
[the remark this answers is not in the document]
Comment (evgenitsn) [reply src/text-field/text-field.tsx:23]: Yup, nice catch. I'll update the according doc pages.

**Review by Vanals — APPROVED** [review]

**Review by mutebg — APPROVED** [review]

File: src/text-field/text-field.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (evgenitsn) [reply src/text-field/text-field.tsx:23]: Done 🚀

**Review by mutebg — APPROVED** [review]

**Review by Xin00163 — APPROVED** [review]

**Pull request merged**